core/templates/dev/head/pages/topic_editor/TopicEditorStateService.js

```diff
@@ -87,6 +111,35 @@ oppia.factory('TopicEditorStateService', [
         });
       },
 
+      /**
+       * Loads, or reloads, the subtopic page stored by this service given a
+       * specified topic ID and subtopic ID.
+       */
+      loadSubtopicPage: function(topicId, subtopicId) {
+        var subtopicPageId = topicId + '-' + subtopicId;
+        if (_getSubtopicPageIndex(subtopicPageId) !== -1) {
+          _subtopicPage = angular.copy(
+            _subtopicPages[_getSubtopicPageIndex(subtopicPageId)]);
+          $rootScope.$broadcast(EVENT_SUBTOPIC_PAGE_LOADED);
+          return;
+        }
+        EditableTopicBackendApiService.fetchSubtopicPage(
+          topicId, subtopicId).then(
+          function(newBackendSubtopicPageObject) {
+            if (newBackendSubtopicPageObject === null) {
```

Comment:
I feel a bit iffy about this. If the subtopic page doesn't exist, why are we trying to fetch it?

Reply:
If it's a newly created subtopic, it wouldn't exist in the datastore, so is the return value is null, I am creating a local subtopic page object and adding it to the array.

Reply:
Ditto, see comment above. If the frontend knows that the subtopic doesn't exist in the backend then it shouldn't even request it.

Reply:
Done.